test/test_subdomains.py

```diff
@@ -135,3 +135,26 @@ def test_ValueError_rasied_when_volume_ids_are_not_unique():
 
     with pytest.raises(ValueError, match="Volume ids are not unique"):
         my_test_model.define_markers_and_measures()
+
+
+def test_ValueError_raised_when_id_not_found_in_surface_subdomains():
+    """test that a ValueError is raised when an id is not found in the list of surface subdomains"""
+
+    surfaces = [F.SurfaceSubdomain(id=1)]
+
+    with pytest.raises(ValueError, match="id 3 not found in list of surfaces"):
+        F.find_surface_from_id(3, surfaces)
+
+
+@pytest.mark.parametrize("input", [1, 2, 3, 4])
```

Comment:
maybe no need for a parametrized test here as a simple for loop is enough.

Plus if would be interesting to test a case where the subdomain ids in `surfaces` are not ordered like:

```surfaces = [ surf_2, surf_1,surf_4, surf_3]```